grib/src/test/java/ucar/nc2/grib/coord/TestDiscontiguousInterval.java

```diff
@@ -0,0 +1,50 @@
+package ucar.nc2.grib.coord;
+
+import org.junit.Assert;
+import org.junit.Test;
+import ucar.ma2.Array;
+import ucar.nc2.*;
+
+import java.io.IOException;
+
+public class TestDiscontiguousInterval {
+
+  @Test
+  public void testTimeCoordinate1D_isMonotonicallyIncreasing() throws IOException {
+    String testfile = "../grib/src/test/data/GFS_Global_onedeg_20220627_0000.precipOnly.grib2";
+    try (NetcdfFile nc = NetcdfFiles.open(testfile)) {
+      Variable dataVar = nc.findVariable("Total_precipitation_surface_Mixed_intervals_Accumulation");
+      Assert.assertNotNull(dataVar);
+
+      Dimension timeDim = null;
+      for (Dimension dim : dataVar.getDimensions()) {
+        if (dim.getShortName().startsWith("time")) {
+          timeDim = dim;
+          break;
+        }
+      }
+      Variable timeCoordVar = nc.findVariable(timeDim.getShortName());
+      Assert.assertNotNull(timeCoordVar);
+
+      Attribute att = timeCoordVar.findAttribute("bounds");
+      Assert.assertNotNull(att);
+      Assert.assertEquals(timeDim.getShortName() + "_bounds", att.getStringValue());
+
+      Array timeCoordValues = timeCoordVar.read();
+      checkTimeCoordinateVariable1D_IsMonotonicallyIncreasing(timeDim.getLength(), timeCoordValues);
+    }
+  }
+
+  private void checkTimeCoordinateVariable1D_IsMonotonicallyIncreasing(int timeDimLength, Array timeCoordValues) {
+    double currentValue = timeCoordValues.getDouble(0);
+    double prevValue = currentValue;
+    StringBuilder valuesSoFar = new StringBuilder();
+    valuesSoFar.append(currentValue);
+    for (int i = 1; i < timeDimLength; i++) {
+      currentValue = timeCoordValues.getDouble(i);
+      valuesSoFar.append(", ").append(currentValue);
+      Assert.assertTrue("Not monotonically increasing: [" + valuesSoFar + "]", currentValue > prevValue);
```

Comment:
for monotonic, shouldn't it be equality be allowed, I mean like:
```suggestion
      Assert.assertTrue("Not monotonically increasing: [" + valuesSoFar + "]", currentValue >= prevValue);
```

Reply:
Interesting. The [NUG defines "coordinate variable" including this phrase] (https://docs.unidata.ucar.edu/nug/current/netcdf_data_set_components.html#coordinate_variables) "strictly monotonic (all values are different and either increasing or decreasing)". While [CF says](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#terminology) it uses the term "coordinate variable" precisely as defined in the NUG but then drops "strictly" in the text.

I believe coordinate variables need to be "strictly monotonic" to support transformations between coordinate systems and the like. So I think the ">" is correct. I will add "strictly" to the language in this PR to clarify. I also think I will open issues on CF to update the language so that it is clear that values must be unique.

Reply:
I think I addressed this with latest commit. So will mark resolved.